lib/patcher.ts

```diff
@@ -172,6 +172,67 @@ export function getReprinter(path: any) {
   const lines = origLoc && origLoc.lines;
   const reprints: any[] = [];
 
+  let     parentNode      = path.getParentNode ();
+  let     origParentNode  = parentNode?.original;
+
+  if ( (node as any).type === 'Identifier' && orig &&
+       parentNode?.type === 'CallExpression' &&
+       origParentNode?.type === 'CallExpression' ) {
+      if (path.getName () !== 'callee') {
+          // console.log (`- got argument`);
+          let     origIndex   = origParentNode.arguments.findIndex ((n: any) => n === orig);
+
+          if (origIndex !== -1) {
+              // Let's process starting point
+              let     origStart       = origLoc.start;
+              let     origEnd         = origLoc.end;
+              let     startColumn     = origStart.column;
+              let     endColumn       = origEnd.column;
+
+              if ((node as any).comments) {
+                  let     lastComment     = (node as any).comments.at (-1);
+                  let     commentsEnd     = lastComment.loc.end;
+
+                  // only process same-line situations
+                  // Also, assuming leading comments
+                  if ( origStart.line === commentsEnd.line &&
+                       lastComment.leading ) {
+                          startColumn     = commentsEnd.column;
+                          // Lets take a ride until non-whitespace is encountered
+                          while (origLoc.lines.charAt ({line: origStart.line, column: startColumn}) === ' ') {
+                              startColumn ++;
+                          }
+                  }
+              }
+              // else if (origIndex === 0) {
```

Comment:
What purpose of this commented code?

Reply:
this is not a completed implementation, as not all cases are handled. However cases that are indeed handled should work as expected.

Basic idea is this: let's be greedy and find the location of the semantic element before argument `a`, in `testFunc (/** @type */ (a), b)`, in order to grep the whole `(a)` lot.

if `Identifier` a has leading comments, let's find till `*/`;
else if a is the first argument, let's find the place of the opening parenthesis (to answer your question)
else let's find the place of the previous comma Token.

Same logic goes to finding the last position of the `(a)` whole lot.

As for the failed tests on NodeJS 12 and 14, its due to `comments.at (-1)` usage. It's the same as `comments[comments.length-1]`. However, I think it should be handled by tsconfig for transpiling, not the code itself.